ReactiveXComponent/RabbitMq/RabbitMqSubscriber.cs

```diff
@@ -0,0 +1,237 @@
+            _subscribers.TryGetValue(subscriberKey, out rabbitMqSubscriberInfos);
+            if (rabbitMqSubscriberInfos == null) return;
+            try
+            {
+                rabbitMqSubscriberInfos.Subscriber.Received += (o, e) =>
+                {
+                    rabbitMqSubscriberInfos.Channel?.BasicAck(e.DeliveryTag, false);
+                    DispatchMessage(e);
+                };
+            }
+            catch (EndOfStreamException ex)
+            {
+                ConnectionFailed?.Invoke(this, "Subscriber has been interrupted : " + ex.Message);
+            }    
+        }
+
+        private void AddListenerToRepository(SubscriberKey subscriberKey, Action<MessageEventArgs> stateMachineListener)
+        {
+            _listenerBySubscriberKeyRepo.AddOrUpdate(subscriberKey, 
+            key => new ThreadSafeList<Action<MessageEventArgs>> { stateMachineListener },
+            (key, oldValue) =>
+            {
+                oldValue.Add(stateMachineListener);
+                return oldValue;
+            });
+        }
+
+        public event EventHandler<string> ConnectionFailed;
+
+        private void ChannelOnModelShutdown(object sender, ShutdownEventArgs shutdownEventArgs)
+        {
+            ConnectionFailed?.Invoke(this, shutdownEventArgs.ReplyText);
+        }
+
+        private void DeleteSubscription(SubscriberKey subscriberkey)
+        {
+            RabbitMqSubscriberInfos rabbitMqSubscriberInfos;
+            _subscribers.TryRemove(subscriberkey, out rabbitMqSubscriberInfos);
+            if (rabbitMqSubscriberInfos == null) return;
+            rabbitMqSubscriberInfos.Channel.ModelShutdown -= ChannelOnModelShutdown;
+            rabbitMqSubscriberInfos.Subscriber.OnCancel();
+            rabbitMqSubscriberInfos.Channel.Close();
+        }
+
+        private void DispatchMessage(BasicDeliverEventArgs basicAckEventArgs)
+        {
+            var obj = _serializer.Deserialize(new MemoryStream(basicAckEventArgs.Body));
+            var msgEventArgs = new MessageEventArgs(
+                RabbitMqHeaderConverter.ConvertStateMachineRef(basicAckEventArgs.BasicProperties.Headers),
+                obj);
+            OnMessageReceived(msgEventArgs);
+        }
+
+        private void OnMessageReceived(MessageEventArgs e)
+        {
+            MessageReceived?.Invoke(this, e);
+        }
+
+        public void Unsubscribe(string component, string stateMachine, Action<MessageEventArgs> stateMachineListener)
+        {
+            var subscriberKey = new SubscriberKey(_xcConfiguration.GetComponentCode(component), _xcConfiguration.GetStateMachineCode(component, stateMachine));
+            DeleteSubscription(subscriberKey);
+            ThreadSafeList<Action<MessageEventArgs>> listStateMachineListener;
+            _listenerBySubscriberKeyRepo.TryGetValue(subscriberKey, out listStateMachineListener);
+            listStateMachineListener?.TryRemove(stateMachineListener);
```

Comment:
A null value test is still present. I would write it this way:

`if (_listenerBySubscriberKeyRepo.TryGetValue(subscriberKey, out listStateMachineListener))
 {
     listStateMachineListener.TryRemove(stateMachineListener);
 }`